From 90c25efdd7ca07f889f191d4da9fdd2b99551819 Mon Sep 17 00:00:00 2001
From: Ben Pfaff <blp@ovn.org>
Date: Fri, 23 Dec 2016 13:33:50 -0800
Subject: [PATCH] tests: Fix race in "ovn -- vtep: 3 HVs, 1 VIFs/HV, 1 GW, 1
 LS".

The 1-second sleep to wait for the bind to occur is not long enough on
slow mips machines.  This fixes the problem.

Reported-at: https://buildd.debian.org/status/fetch.php?pkg=openvswitch&arch=mipsel&ver=2.6.2%7Epre%2Bgit20161223-1&stamp=1482523419&file=log
Signed-off-by: Ben Pfaff <blp@ovn.org>
Acked-by: Justin Pettit <jpettit@ovn.org>
---
 tests/ovn.at | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/tests/ovn.at b/tests/ovn.at
index bdeaa41..f2c941a 100644
--- a/tests/ovn.at
+++ b/tests/ovn.at
@@ -1710,9 +1710,7 @@ vtep-ctl add-ls lsw0
 start_daemon ovs-vtep br-vtep
 start_daemon ovn-controller-vtep --vtep-db=unix:"$ovs_base"/vtep/db.sock --ovnsb-db=unix:"$ovs_base"/ovn-sb/ovn-sb.sock
 
-sleep 1
-
-vtep-ctl bind-ls br-vtep br-vtep_n2 0 lsw0
+OVS_WAIT_UNTIL([vtep-ctl bind-ls br-vtep br-vtep_n2 0 lsw0])
 
 OVS_WAIT_UNTIL([test -n "`as vtep vtep-ctl get-replication-mode lsw0 |
                grep -- source`"])
-- 
2.9.3

